src/test/java/com/google/sps/servlets/LoadGamePageServletTest.java

```diff
@@ -0,0 +1,119 @@
+
+        responseWriter = new StringWriter();
+        when(response.getWriter()).thenReturn(new PrintWriter(responseWriter));
+
+        servlet = new LoadGamePageServlet();
+    }
+
+    /**
+    * Gets rid of local datastore after each test.
+    */
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    /**
+    * Tests that the game information is correctly loaded.
+    */
+    @Test
+    public void testLoadExisitingGame() throws Exception {
+        Game.Builder gamebuilder = new Game.Builder("abcGameId", "abcGameName");
+        gamebuilder.setGameCreator("abcUserId").setGameDescription("abcGameDescription").setStages(new ArrayList<String>(Arrays.asList("stage1id", "stage2id")));
+        gamebuilder.setNumTimesPlayed(10).setNumTimesFinished(5).setNumStarVotes(12).setTotalStars(48).setNumDifficultyVotes(12).setTotalDifficulty(24);
+        datastoreManager.createOrReplaceGame(gamebuilder.build());
+
+        Stage.Builder stagebuilder = new Stage.Builder("stage1id", 1);
+        stagebuilder.setKey("abc1k").setStartingHint("abc1h").setStartingLocation(new Coordinates(0.0, 0.0));
+        Hint stage1hint1 = new Hint.Builder("stage1hint1id", 1).setLocation(new Coordinates(0.0, 0.0)).setText("abc11").build();
+        Hint stage1hint2 = new Hint.Builder("stage1hint2id", 2).setLocation(new Coordinates(0.0, 0.0)).setText("abc12").build();
+        stagebuilder.setHints(new ArrayList<Hint>(Arrays.asList(stage1hint1, stage1hint2)));
+        datastoreManager.createOrReplaceStage(stagebuilder.build());
+
+        stagebuilder = new Stage.Builder("stage2id", 2);
+        stagebuilder.setKey("abc2k").setStartingHint("abc2h").setStartingLocation(new Coordinates(1.0, 1.0));
+        Hint stage2hint1 = new Hint.Builder("stage2hint1id", 1).setLocation(new Coordinates(0.0, 0.0)).setText("abc21").build();
+        stagebuilder.setHints(new ArrayList<Hint>(Arrays.asList(stage2hint1)));
+        datastoreManager.createOrReplaceStage(stagebuilder.build());
+
+        User.Builder userbuilder = new User.Builder("abcUserId").setUsername("abcUsername");
+        datastoreManager.createOrReplaceUser(userbuilder.build());
+
+        servlet.doPost(request, response);
+        String json = responseWriter.toString();
+        Type ReducedGameDataType = new TypeToken<ReducedGameData>(){}.getType();
+        ReducedGameData res = new Gson().fromJson(json, ReducedGameDataType);
+        
+        Assert.assertTrue(res.gameID.equals("abcGameId"));
+        Assert.assertTrue(res.gameName.equals("abcGameName"));
+        Assert.assertTrue(res.gameDescription.equals("abcGameDescription"));
+        Assert.assertTrue(res.creatorUsername.equals("abcUsername"));
+        Assert.assertTrue(res.difficulty == 2.0);
+        Assert.assertTrue(res.stars == 4.0);
+        Assert.assertTrue(res.stageLocations.equals(new ArrayList<Coordinates>(Arrays.asList(new Coordinates(0.0, 0.0), new Coordinates(1.0, 1.0)))));
```

Comment:
at this point you have done `new Coordinates(0.0, 0.0)` 5 times, maybe consider putting this into a variable instead.

Reply:
I don't think this is a good idea because if I added a variable just for this, I'd have to worry about using that variable whenever I want to use the specific coordinate (0,0), which seems a bit overkill. In my opinion, `new Coordinates(0,0)` is short and clear enough so that I can just keep using it every time.